src/ripple/overlay/MessageScheduler.md

```diff
@@ -0,0 +1,350 @@
+# MessageScheduler
+
+At times, our server needs to communicate with its peers:
+sending them transactions, proposals, and validations, or requesting data.
+We need a component to facilitate this communication that balances concerns
+including throughput, congestion, and prioritization.
+
+## Motivation
+
+The existing architecture for communication is designed for this pattern:
+
+- Each sender builds a fixed size `PeerSet` holding a subset of connected
+    peers and sends messages to those peers whenever they want.
+- Each receiver is a singleton and all responses of a given message type are
+    delivered to the same receiver.
```

Comment:
Sorry.  I'm at the very top of the document and I'm already getting confused with the terminology.  I'm misunderstanding what a `sender` and a `receiver` are.

- When I see `sender` I think that corresponds to a rippled instance.  That aligns with my understanding of rippled and the `peers` response in rippled's `server_info` call.

- But then I see "receiver is a singleton" and I know that I'm off base.  A rippled instance cannot possibly meet the description of `receiver` in that paragraph.

I think I need more context.  Can you give concrete examples of the things you are calling `sender` and `receiver`?  Since you are describing the existing architecture there should be some existing examples.